javascript/ql/lib/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -80,499 +132,596 @@ module TaintedPath {
+    class ResolvingPathCall extends PathTransformationNode {
+      DataFlow::Node input;
+      DataFlow::Node output;
+
+      ResolvingPathCall() {
+        this = NodeJSLib::Path::moduleMember("resolve").getACall() and
+        input = this.getAnArgument() and
+        output = this
+        or
+        this = NodeJSLib::FS::moduleMember("realpathSync").getACall() and
+        input = this.getArgument(0) and
+        output = this
+        or
+        this = NodeJSLib::FS::moduleMember("realpath").getACall() and
+        input = this.getArgument(0) and
+        output = this.getCallback(1).getParameter(1)
+      }
 
-    /**
-     * Gets the normalized path.
-     */
-    DataFlow::Node getOutput() { result = output }
-  }
+      override predicate step(
+        Path::PathLabel inputLabel, DataFlow::Node inputNode, Path::PathLabel outputLabel,
+        DataFlow::Node outputNode
+      ) {
+        input = inputNode and
+        output = outputNode and
+        outputLabel.getPlatform() = inputLabel.getPlatform() and
+        outputLabel.isAbsolute() and
+        outputLabel.isNormalized()
+      }
 
-  /**
-   * A call that normalizes a path and converts it to a relative path.
-   */
-  class NormalizingRelativePathCall extends DataFlow::CallNode {
-    DataFlow::Node input;
-    DataFlow::Node output;
+      /**
+       * Returns the input node from which steps are taken.
+       */
+      DataFlow::Node getInput() { result = input }
 
-    NormalizingRelativePathCall() {
-      this = NodeJSLib::Path::moduleMember("relative").getACall() and
-      input = this.getAnArgument() and
-      output = this
+      /**
+       * Returns the output node to which steps are taken.
+       */
+      DataFlow::Node getOutput() { result = output }
     }
 
     /**
-     * Gets the input path to be normalized.
+     * A call to the path.relative method.
      */
-    DataFlow::Node getInput() { result = input }
+    class RelativePathCall extends PathTransformationNode {
+      RelativePathCall() { this = NodeJSLib::Path::moduleMember("relative").getACall() }
+
+      override predicate step(
+        Path::PathLabel inputLabel, DataFlow::Node inputNode, Path::PathLabel outputLabel,
+        DataFlow::Node outputNode
+      ) {
+        this.getAnArgument() = inputNode and
+        this = outputNode and
+        outputLabel.getPlatform() = inputLabel.getPlatform() and
+        (
+          inputLabel.getPlatform() = Path::platformPosix() and
+          outputLabel = inputLabel.toRelative().toNormalized()
+          or
+          // relative, on win32, does not always return a normalized path, eg.,
+          // when both arguments are equal, relative will return ''
+          // (normalized would be '.').
+          // It also does not always return a relative path, eg.,
+          // relative('C:\\', 'A:\\') returns 'A:\\'.
+          inputLabel.getPlatform() = Path::platformWin32() and
+          outputLabel = inputLabel
```

Comment:
I think you could move all of these windows specific paths into a windows specific file (`TaintedWindowsPathQuery.qll`?).    
In all cases you can do something like this (the name is bad):   
```CodeQL
class RelativePathCallWindows extends RelativePathCall {
  override predicate step(...) {
    super.step(...)
    or
    this.getAnArgument() = inputNode and 
    this = outputNode and
    outputLabel.getPlatform() = inputLabel.getPlatform() and
    inputLabel.getPlatform() = Path::platformWin32() and
    outputLabel = inputLabel
  }
}
```

If too much logic ends up getting duplicated, then you can move some if it into a utility predicate on the class.